src/app/shared/components/hmcts-em-viewer-ui/data/pdf-render.service.spec.ts

```diff
@@ -0,0 +1,104 @@
+import { PdfRenderService } from './pdf-render.service';
+import { ElementRef } from '@angular/core';
+import { TestBed, inject } from '@angular/core/testing';
+import { RenderOptions } from './js-wrapper/renderOptions.model';
+import { EmLoggerService } from '../logging/em-logger.service';
+import { PdfWrapper } from './js-wrapper/pdf-wrapper';
+import { PdfAnnotateWrapper } from './js-wrapper/pdf-annotate-wrapper';
+import { BehaviorSubject } from 'rxjs';
+
+class MockPdfAnnotateWrapper {
+    renderPage(pageNumber) {}
+    enableRect(tool) {}
+    disableRect() {}
+    createPage() {}
+}
+
+class MockPdfWrapper {
+    workerSrc() {}
+    getDocument(documentId) {}
+  }
+  
+describe('PdfService', () => {
+    const mockPdfAnnotateWrapper = new MockPdfAnnotateWrapper();
+    const mockPdfWrapper = new MockPdfWrapper();
+  
+    beforeEach(() => {
+      TestBed.configureTestingModule({
+        providers: [
+          EmLoggerService,
+          PdfRenderService,
+          { provide: PdfAnnotateWrapper, useFactory: () => mockPdfAnnotateWrapper },
+          { provide: PdfWrapper, useFactory: () => mockPdfWrapper }
+        ]
+      });
+    });
+  
+    describe('constructor', () => {
+      it('should be created', inject([PdfRenderService], (service: PdfRenderService) => {
+        expect(service).toBeTruthy();
+      }));
+    });
+
+
+  describe('render', () => {
+    it('render should set workerSrc', inject([PdfRenderService], (service: PdfRenderService) => {
```

Comment:
I'm not sure this is clear as I don't know what is meant by "render should set workerSrc"